### PR TITLE
Add table aws_dax_parameter_group closes #1424

### LINKED ISSUE #1424: Add table aws_dax_parameter_group
**References**
[Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dax_parameter_group)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_dax_parameter_group []

PRETEST: tests/aws_dax_parameter_group

TEST: tests/aws_dax_parameter_group
Running terraform
data.aws_region.alternate: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-2]
data.aws_region.primary: Reading...
data.aws_caller_identity.current: Reading...
data.aws_partition.current: Reading...
data.aws_region.primary: Read complete after 0s [id=us-east-1]
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_caller_identity.current: Read complete after 2s [id=986502156756]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_dax_parameter_group.named_test_resource will be created
  + resource "aws_dax_parameter_group" "named_test_resource" {
      + description = "turbottest1549"
      + id          = (known after apply)
      + name        = "turbottest1549"

      + parameters {
          + name  = "query-ttl-millis"
          + value = "100000"
        }
      + parameters {
          + name  = "record-ttl-millis"
          + value = "100000"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id    = "986502156756"
  + aws_partition = "aws"
  + aws_region    = "us-east-1"
  + resource_aka  = "arn:aws:dax:us-east-1::parametergroup:turbottest1549"
  + resource_id   = (known after apply)
  + resource_name = "turbottest1549"
aws_dax_parameter_group.named_test_resource: Creating...
aws_dax_parameter_group.named_test_resource: Creation complete after 3s [id=turbottest1549]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 44, in data "null_data_source" "resource":
  44: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "986502156756"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:dax:us-east-1::parametergroup:turbottest1549"
resource_id = "turbottest1549"
resource_name = "turbottest1549"

Running SQL query: test-list-query.sql
[
  {
    "description": "turbottest1549",
    "parameter_group_name": "turbottest1549",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

POSTTEST: tests/aws_dax_parameter_group

TEARDOWN: tests/aws_dax_parameter_group

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>

```

> select
  p.parameter_group_name,
  c.cluster_name,
  c.node_type,
  c.status
from
  aws_dax_parameter_group as p,
  aws_dax_cluster as c
where
  c.parameter_group ->> 'ParameterGroupName' = p.parameter_group_name;
+----------------------+--------------+--------------+----------+
| parameter_group_name | cluster_name | node_type    | status   |
+----------------------+--------------+--------------+----------+
| test3                | test53       | dax.r5.large | creating |
+----------------------+--------------+--------------+----------+
> select
  parameter_group_name,
  description,
  region
from
  aws_dax_parameter_group;
+----------------------+------------------------------------+----------------+
| parameter_group_name | description                        | region         |
+----------------------+------------------------------------+----------------+
| default.dax1.0       | Default parameter group for dax1.0 | ap-south-1     |
| default.dax1.0       | Default parameter group for dax1.0 | ap-southeast-1 |
| default.dax1.0       | Default parameter group for dax1.0 | ap-northeast-1 |
| default.dax1.0       | Default parameter group for dax1.0 | ap-southeast-2 |
| default.dax1.0       | Default parameter group for dax1.0 | eu-west-2      |
| default.dax1.0       | Default parameter group for dax1.0 | eu-central-1   |
| default.dax1.0       | Default parameter group for dax1.0 | eu-west-1      |
| default.dax1.0       | Default parameter group for dax1.0 | eu-west-3      |
| default.dax1.0       | Default parameter group for dax1.0 | us-west-2      |
| test3                | testing                            | us-east-1      |
| default.dax1.0       | Default parameter group for dax1.0 | us-east-2      |
| default.dax1.0       | Default parameter group for dax1.0 | us-east-1      |
| default.dax1.0       | Default parameter group for dax1.0 | sa-east-1      |
| default.dax1.0       | Default parameter group for dax1.0 | us-west-1      |
+----------------------+------------------------------------+----------------+

> select parameter_group_name, jsonb_pretty(parameters) from aws_dax_parameter_group
+----------------------+-----------------------------------------------------------------------------------------------------------------+
| parameter_group_name | jsonb_pretty                                                                                                    |
+----------------------+-----------------------------------------------------------------------------------------------------------------+
| default.dax1.0       | [                                                                                                               |
|                      |     {                                                                                                           |
|                      |         "Source": "user",                                                                                       |
|                      |         "DataType": "integer",                                                                                  |
|                      |         "ChangeType": "IMMEDIATE",                                                                              |
|                      |         "Description": "Duration in milliseconds for queries to remain cached",                                 |
|                      |         "IsModifiable": "TRUE",                                                                                 |
|                      |         "AllowedValues": "0-",                                                                                  |
|                      |         "ParameterName": "query-ttl-millis",                                                                    |
|                      |         "ParameterType": "DEFAULT",                                                                             |
|                      |         "ParameterValue": "300000",                                                                             |
|                      |         "NodeTypeSpecificValues": [                                                                             |
|                      |         ]                                                                                                       |
|                      |     },                                                                                                          |
|                      |     {                                                                                                           |
|                      |         "Source": "user",                                                                                       |
|                      |         "DataType": "integer",                                                                                  |
|                      |         "ChangeType": "IMMEDIATE",                                                                              |
|                      |         "Description": "Duration in milliseconds for records to remain valid in cache (Default: 0 = infinite)", |
|                      |         "IsModifiable": "TRUE",                                                                                 |
|                      |         "AllowedValues": "0-",                                                                                  |
|                      |         "ParameterName": "record-ttl-millis",                                                                   |
|                      |         "ParameterType": "DEFAULT",                                                                             |
|                      |         "ParameterValue": "300000",                                                                             |
|                      |         "NodeTypeSpecificValues": [                                                                             |
|                      |         ]                                                                                                       |
|                      |     }                                                                                                           |
|                      | ]                                                                                                               |
| default.dax1.0       | [                                                                                                               |
|                      |     {                                                                                                           |
|                      |         "Source": "user",                                                                                       |
|                      |         "DataType": "integer",                                                                                  |
|                      |         "ChangeType": "IMMEDIATE",                                                                              |
|                      |         "Description": "Duration in milliseconds for queries to remain cached",                                 |
|                      |         "IsModifiable": "TRUE",                                                                                 |
|                      |         "AllowedValues": "0-",                                                                                  |
|                      |         "ParameterName": "query-ttl-millis",                                                                    |
|                      |         "ParameterType": "DEFAULT",                                                                             |
|                      |         "ParameterValue": "300000",                                                                             |
|                      |         "NodeTypeSpecificValues": [                                                                             |
|                      |         ]                                                                                                       |
|                      |     },                                                                                                          |
|                      |     {                                                                                                           |
|                      |         "Source": "user",                                                                                       |
|                      |         "DataType": "integer",                                                                                  |
|                      |         "ChangeType": "IMMEDIATE",                                                                              |
|                      |         "Description": "Duration in milliseconds for records to remain valid in cache (Default: 0 = infinite)", |
|                      |         "IsModifiable": "TRUE",                                                                                 |
|                      |         "AllowedValues": "0-",                                                                                  |
|                      |         "ParameterName": "record-ttl-millis",                                                                   |
|                      |         "ParameterType": "DEFAULT",                                                                             |
|                      |         "ParameterValue": "300000",                                                                             |
|                      |         "NodeTypeSpecificValues": [                                                                             |
|                      |         ]                                                                                                       |
|                      |     }                                                                                                           |
|                      | ]                                                                                                               |

```
</details>

### CODE REVIEW

File: aws/table_aws_dax_parameter_group.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+				Name:        "parameter_group_name",
+				Description: "The name of the parameter group.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "description",
+				Description: "A description of the parameter group.",
+				Type:        proto.ColumnType_STRING,
+			},
+
```
Comment (bigdatasourav) [inline aws/table_aws_dax_parameter_group.go:42]: Please add parameters details - Separate hydrate column. API - DescribeParameters

**Review by bigdatasourav — APPROVED** [review]
LGTM

File: docs/tables/aws_dax_parameter_group.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+```sql
+select
+  parameter_group_name,
+  description,
+  region
+from
+  aws_dax_parameter_group;
+```
+
+### Get cluster details that associated with parameter group
```
Comment (misraved) [inline docs/tables/aws_dax_parameter_group.md:18]:
```suggestion
### Get cluster details associated with the parameter group
```

File: aws/table_aws_dax_parameter_group.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "parameter_group_name",
+				Description: "The name of the parameter group.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "description",
+				Description: "A description of the parameter group.",
```
Comment (misraved) [inline aws/table_aws_dax_parameter_group.go:39]:
```suggestion
				Description: "Description of the parameter group.",
```

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**